pullman/storageproviders/http/downloader.go

```diff
@@ -75,14 +78,37 @@ func (c *httpFetcher) download(ctx context.Context, req *http.Request, filename
 	}
 	defer resp.Body.Close()
 
-	file, fileErr := pullman.OpenFile(filename)
-	if fileErr != nil {
-		return fmt.Errorf("unable to open local file '%s' for writing: %w", filename, fileErr)
+	if resp.StatusCode != http.StatusOK {
+		body, readErr := io.ReadAll(resp.Body)
+		if readErr != nil {
+			return fmt.Errorf("error getting resource '%s'. HTTP Status Code: %d, Error: %w", req.URL.String(), resp.StatusCode, readErr)
+		}
+		return fmt.Errorf("error getting resource '%s'. HTTP Status Code: %d, Error: %s", req.URL.String(), resp.StatusCode, string(body))
 	}
-	defer file.Close()
 
-	if _, err = io.Copy(file, resp.Body); err != nil {
-		return fmt.Errorf("error writing resource to local file '%s': %w", filename, err)
+	contentType := resp.Header.Get("Content-type")
+	if strings.Contains(contentType, "application/x-tar") || strings.Contains(contentType, "application/x-gtar") ||
+		strings.Contains(contentType, "application/x-gzip") || strings.Contains(contentType, "application/gzip") {
```

Comment:
There is [also](https://superuser.com/a/960710) `application/tar` and `application/tar+gzip` (though I guess `Contains` would catch the latter).

Also wondering whether we should detect based on file extensions in the URL even if `Content-Type` isn't specified?

Reply:
Sure, I can add file extension checks as fall backs.